File: BE-MyCarMaster/src/main/java/softeer/be_my_car_master/domain/engine/Engine.java
package softeer.be_my_car_master.domain.engine;


import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import softeer.be_my_car_master.global.annotation.Domain;

@Domain
@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Engine {

	private Long id;
	private String name;
	private String description;
	private Integer ratio;
	private Integer price;
	private String imgUrl;
	private Double fuelMin;
	private Double fuelMax;
	private Integer power;
	private Double toque;

	public boolean isRightEngine(Long engineId, Integer enginePrice) {
		return engineId.equals(id) && enginePrice.equals(price);
	}
}

File: iOS-MyCarMaster/MyCarMaster/MyCarMaster/Model/Entity/Engine.swift
//
//  Engine.swift
//  MyCarMaster
//
//  Created by SEUNGMIN OH on 2023/08/12.
//

import Foundation

struct Engine: Hashable, Priceable {
    var model: String
    var id: Int
    var name: String
    var ratio: Int
    var description: String
    var fuelMin: Double
    var fuelMax: Double
    var power: Int
    var toque: Double
    var price: Int
    var imageURL: URL?
}
